lib/ui/window.dart

```diff
@@ -172,18 +172,14 @@ enum AppLifecycleState {
   ///
   /// When the application is in this state, the engine will not call the
   /// [Window.onBeginFrame] and [Window.onDrawFrame] callbacks.
-  ///
-  /// Android apps in this state should assume that they may enter the
-  /// [suspending] state at any time.
   paused,
 
-  /// The application will be suspended momentarily.
-  ///
-  /// When the application is in this state, the engine will not call the
-  /// [Window.onBeginFrame] and [Window.onDrawFrame] callbacks.
+  /// The application is detached from view.
   ///
-  /// On iOS, this state is currently unused.
-  suspending,
```

Comment:
@cbracken 
It doesn't seem like we are still using this enum any where in our code. Since this pr will be a breaking change anyway. Is it ok to remove this?

Reply:
Is there any reason not to deprecate for one stable release , document as unused, then delete it in order to make the transition slightly less painful to users?

Reply:
I think it is less painful to remove the enum and add the new enum in one pr. If you think it is better to add a document warning ahead of time, I can do a document only change pr first and wait for some release cycle before merge this one

Reply:
I'm for doing it in one PR as well. Adding an enum value is already a non-gracefully-failable breaking change. We already can't make a soft transition.

Reply:
This enum is still used in the framework. It looks like https://github.com/flutter/flutter/pull/39945 will remove it (from scheduler/binding.dart), but that PR hasn't been merged yet. As a result, this PR will block engine rolls moving forward.